metaci/build/tests/test_flows.py

```diff
@@ -0,0 +1,209 @@
+        # copy sample test file to output.xml file being used in the test
+        copy_file(
+            (TEST_ROBOT_OUTPUT_FILES / "robot_1.xml"), (output_dir / "output.xml"),
+        )
+
+        step_spec = get_spec("1", name="Robot", cls=Robot)
+        step_result = StepResult(
+            step_num="1",
+            task_name="Robot",
+            path="Robot",
+            result="something",
+            return_values={"robot_outputdir": str(output_dir)},
+            exception=None,
+        )
+
+        build_flow = BuildFlowFactory()
+        metaci_callbacks = MetaCIFlowCallback(build_flow.id)
+        metaci_callbacks.post_task(step_spec, step_result)
+
+        # For output_1 we should have a single BuildFlowAsset,
+        # a single TestResult, and no TestResultAssets (screenshots)
+        assert (
+            1
+            == BuildFlowAsset.objects.filter(
+                category="robot-output", build_flow=build_flow
+            ).count()
+        )
+        assert 1 == TestResult.objects.all().count()
+        assert 0 == TestResultAsset.objects.all().count()
+
+        step_spec = get_spec("2", name="Robot", cls=Robot)
+        step_result = StepResult(
+            step_num="2",
+            task_name="Robot",
+            path="Robot",
+            result="something",
+            return_values={"robot_outputdir": str(output_dir)},
+            exception=None,
+        )
+
+        # create screenshots for the robot_screenshots.xml
+        open((output_dir / "selenium-screenshot-1.png"), mode="w+")
+        open((output_dir / "selenium-screenshot-2.png"), mode="w+")
+        # copy sample test file to output.xml file being used in the test
+        copy_file(
+            (TEST_ROBOT_OUTPUT_FILES / "robot_screenshots.xml"),
+            (output_dir / "output.xml"),
+        )
+
+        metaci_callbacks.post_task(step_spec, step_result)
+
+        output_files = BuildFlowAsset.objects.filter(
+            category="robot-output", build_flow=build_flow
+        )
+        # There should now be two output files,
+        # one for each Robot task that has executed
+        assert len(output_files) == 2
+
+        # There should be one screenshot created during suite setup
+        assert 1 == BuildFlowAsset.objects.filter(category="robot-screenshot").count()
+
+        # No screenshots created for 'Via API' test
+        tr_method = TestMethod.objects.get(name="Via API")
+        test_api = TestResult.objects.get(method=tr_method)
+        assert 0 == test_api.assets.count()
+
+        # One screenshot created for 'Via UI' test
+        tr_method = TestMethod.objects.get(name="Via UI")
+        test_ui = TestResult.objects.get(method=tr_method)
+        assert 1 == test_ui.assets.count()
+
+        # Three tests total between the two output files
+        assert 3 == TestMethod.objects.all().count()
+
+
+def copy_file(input_file, output_file):
+    """Copy contents of input_file to output_file"""
```

Comment:
You can use shutil.copyfile for this: https://docs.python.org/3/library/shutil.html#shutil.copyfile